### DISCUSSION TITLE
'$this->record' null when creating resource in modal mode

### DISCUSSION THREAD

**Discussion by vmosoti** [filamentphp/filament#5537] [Help]
Inside the `after` call back after inserting a record in the database, the `$this->record ` is null

```php
        return [
            Actions\CreateAction::make()->after(function () {
                
                dd($this->record); // null
            }),
        ];
```

Any one to point me where I could be doing it wrong?

**Comment by danharrin** [comment]
Try this:

```php
        return [
            Actions\CreateAction::make()->after(function ($record) {
                
                dd($record);
            }),
        ];
```

**Reply by vmosoti** [reply]
Hi @danharrin and thanks for your reply. This still returns null

**Reply by danharrin** [reply]
That should work, so please open an issue with a reproduction repository

**Reply by alepieser** [reply]
Hello @danharrin, we are facing the same issue. There is an issue open or alternative ?

**Reply by danharrin** [reply]
please open an issue

**Reply by alepieser** [reply]
Done ! #5844
